src/index/TextIndexReadWrite.h

```diff
@@ -158,12 +154,13 @@ class FrequencyEncode {
   using CodeMap = TypedMap;
   using CodeBook = TypedVector;
 
-  explicit FrequencyEncode(const TypedVector& vectorToEncode);
+  // View must be an input range with value type `T`.
+  template <typename View>
+  explicit FrequencyEncode(View&& view);
```

Comment:
Making this compatible with the C++17 backports AND using constraints/concepts would be somewhat involved here. I have simply decided to not use concepts here, as we only use this in a single place.